### PR TITLE
VHD build: multiple windows version support

### PR DESCRIPTION
**Pull request by mainred**
This PR introduces:
1. sig mode to push image to self defined SIG and default mode to push images to storage account. This have already been achieved by Linux counterpart.
2. multiple windows versions support for VHD building

### CODE REVIEW

**Comment by mainred** [comment]
default mode test: https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=34238006&view=results
sig node test: https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=34231630&view=results

File: vhdbuilder/packer/init-variables.sh [hunk 1]
```diff
@@ -83,6 +83,31 @@ else
...
+# because a pr gives a better chance to take a review of the version changes.
+WINDOWS_IMAGE_SKU=""
+WINDOWS_IMAGE_VERSION=""
+if [ ! -z "${WINDOWS_SKU}" ]; then
+	case "${WINDOWS_SKU}" in
+	"2019") 
+		WINDOWS_IMAGE_SKU="2019-Datacenter-Core-smalldisk"
+		WINDOWS_IMAGE_VERSION="17763.1282.2006061952"
+		;;
+	"2004")
```
Comment (mainred) [inline vhdbuilder/packer/init-variables.sh:100]: windows 2004 is not fully supported in this PR yet, changes like pause image are not included in this PR.

File: .pipelines/.vsts-vhd-builder-windows.yaml [hunk 2]
```diff
@@ -13,6 +15,23 @@
...
+#    This modes is activated when mode 1 is not activated and by setting env var SIG_GALLERY_NAME/SIG_IMAGE_NAME/SIG_IMAGE_VERSION.
+#    It is similar to mode 1 except it will create a gen1 image and the SIG destination is specified by the provided env var.
+#    This is good for quick testing purpose as you can use the SIG image as soon as the build is completed.
+#    steps:
+#    - Build VHD to a SIG gallery as specified by env var SIG_GALLERY_NAME/SIG_IMAGE_NAME/SIG_IMAGE_VERSION
+#
+# default mode: Hyper-Generation V1 +VHD
+#    This default mode generates VHD and store it in a classic storage account. This is mainly for publishing a new VM image to Platform Image Repository(PIR) and Share Image Gallery.
+#    steps:
+#    - create an VHD in Packer to normal storage account
```
Comment (AbelHu) [inline .pipelines/.vsts-vhd-builder-windows.yaml:29]: What is the normal storage account and Packer storage account?

File: vhdbuilder/packer/init-variables.sh [hunk 3]
```diff
@@ -83,6 +83,31 @@ else
 	echo "Skipping SIG check for $MODE"
 fi
 
+# considerations to also add the windows support here instead of an extra script to initialize windows variables:
+# 1. we can demonstrate the whole user defined parameters all at once
+# 2. help us keep in mind that changes of this variables will influnce both windows and linux VHD building
```
Comment (AbelHu) [inline vhdbuilder/packer/init-variables.sh:88]: typo: influence

File: .pipelines/.vsts-vhd-builder-windows.yaml (hunk 2 above)
Comment (mainred) [reply .pipelines/.vsts-vhd-builder-windows.yaml:29]: copy and paste from [Linux pipeline](https://github.com/Azure/AgentBaker/blob/master/.pipelines/.vsts-vhd-builder.yaml#L22). `normal storage account` equals `Packer storage account`.

**Comment by mainred** [comment]
ping @xuto2  and @AbelHu  to take a look, thanks.

**Pull request merged**